tests/manage/pv_services/pv_encryption/test_restore_snapshot_for_encrypted_or_unencrypted_pvc_to_unencrypted_or_encrypted_sc.py

```diff
@@ -0,0 +1,502 @@
+            snap_obj.md5sum = md5sum_after_io
+            snap_objs.append(snap_obj)
+            log.info("Snapshots created")
+
+        # Verify snapshots are ready and verify if encryption key is created in vault
+        log.info("Verify snapshots are ready")
+        for snap_obj in snap_objs:
+            snap_obj.ocp.wait_for_resource(
+                condition="true",
+                resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE,
+                timeout=180,
+            )
+            snapshot_content = get_snapshot_content_obj(snap_obj=snap_obj)
+            snap_handle = snapshot_content.get().get("status").get("snapshotHandle")
+            if kms_provider == constants.VAULT_KMS_PROVIDER:
+                if kms.is_key_present_in_path(
+                    key=snap_handle, path=self.kms.vault_backend_path
+                ):
+                    log.info(f"Vault: Found key for snapshot {snap_obj.name}")
+                else:
+                    raise ResourceNotFoundError(
+                        f"Vault: Key not found for snapshot {snap_obj.name}"
+                    )
+            snap_handles.append(snap_handle)
+
+        # Restoring snapshots to create new PVCs using unencrypted storageclass
+        log.info("Creating new PVCs from snapshots")
+        restore_pvcs = []
+        for snap_obj in snap_objs:
+            # for storageclass in sc_objs:
+            log.info(f"Creating a PVC from snapshot {snap_obj.name}")
+            restore_pvc_obj = snapshot_restore_factory(
+                snapshot_obj=snap_obj,
+                storageclass=self.sc_obj_without_encryption.name,
+                size=f"{self.pvc_size}Gi",
+                volume_mode=snap_obj.parent_volume_mode,
+                access_mode=snap_obj.parent_access_mode,
+                status="",
+            )
+            restore_pvcs.append(restore_pvc_obj)
+
+            log.info(
+                f"Created PVC {restore_pvc_obj.name} from snapshot {snap_obj.name}."
+                f"Used the storage class {self.sc_obj_without_encryption}"
+            )
+            restore_pvc_obj.md5sum = snap_obj.md5sum
+            failure_str = "cannot create unencrypted volume from encrypted volume"
+            # Confirm that the restored PVCs are in Pending status
+            for restore_pvc_obj in restore_pvcs:
+                helpers.wait_for_resource_state(
+                    resource=restore_pvc_obj,
+                    state=constants.STATUS_PENDING,
+                    timeout=200,
+                )
+                restore_pvc_obj.reload()
+                if failure_str in restore_pvc_obj.describe():
+                    log.info(
+                        f"cannot create unencrypted volume from encrypted volume {snap_obj.name}"
+                    )
+                else:
+                    log.warning(
+                        f"able to create unencrypted volume from encrypted volume {snap_obj.name}"
```

Comment:
IMO, this should be logged as an error instead of a warning

Reply:
Updated to error